src/framework/actions/hand.ts

```diff
@@ -4,6 +4,96 @@ import { GameState } from "../types";
 
 import { makeInstance } from "./helpers";
 
+/**
+ * NOTE: For security purposes, drawing should be handled or confirmed by the
+ * server. Active players' clients should not be allowed to manipulate their
+ * draw order, and the contents/order of the draw pile for other players needs
+ * to be unknown.
+ * NOTE: This draws the top card. For abilities that allow drawing an arbitrary
+ * card from the player's deck, a different function should be used.
+ *
+ * @param $
+ * @param pid
+ */
+export const drawCard = ($: GameState): void => {
```

Comment:
The active player isn't the only player who can draw cards (technician...). This (and `shuffle`, e.g.) should probably take a player ID, or take a `PlayerState` pointer instead of the whole `GameState`.

Reply:
This has been corrected.

Reply:
I didn't call it out explicitly, but we will want this change in discard as well (need to be able to discard cards from opponent's hand, and discardAll doesn't need it but wouldn't hurt to be more explicit)